src/main/fox_versus.c

```diff
@@ -0,0 +1,2018 @@
+    }
+
+    return 0;
+}
+
+s32 func_800C075C(void) {
+    RCP_SetupDL(&gMasterDisp, 0x4E);
+    gDPSetPrimColor(gMasterDisp++, 0, 0, 255, 255, 255, 255);
+    func_800BCEF8(94.0f, 126.0f, 1.0f);
+    return 0;
+}
+
+s32 func_800C07C4(void) {
+    s32 i;
+    s32 stick_y = 0;
+    s32 ret = 0;
+
+    for (i = 0; i < 4; i++) {
+        if (gControllerPlugged[i] == 1) {
+            if (stick_y = gControllerPress[i].stick_y) {
+                break;
+            }
+        }
+    }
+
+    if (stick_y) {
+        if (stick_y > 0) {
+            stick_y = -1;
+        } else {
+            stick_y = 1;
+        }
+
+        if (D_8017879C == 0) {
+            ret = stick_y;
+        }
+        D_801787A0++;
+        D_8017879C = stick_y;
+    } else {
+        D_8017879C = 0;
+        D_801787A0 = 0;
+    }
+
+    // clang-format off
+    if (ret != 0) { Audio_PlaySfx(0x49000002U, &D_800C5D28, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C); }
+    // clang-format on
+
+    return ret;
+}
+
+bool func_800C08D0(void) {
+    f32 temp = 0.0f;
+
+    temp += Math_SmoothStepToF(&D_801787CC, 1.55f, 0.6f, 10.0f, 0.04f);
+    temp += Math_SmoothStepToF(&D_801787D0, 0.82f, 0.6f, 10.0f, 0.04f);
+
+    if (temp != 0.0f) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+void func_800C0970(f32 yScale1, f32 yScale2) {
+    RCP_SetupDL(&gMasterDisp, 0x4E);
+    gDPSetPrimColor(gMasterDisp++, 0, 0, 0, 0, 0, 255);
+    func_800BCFFC(98.0f, 72.0f, 5.19f, yScale1);
+    func_800BCFFC(98.0f, 152.0f, 5.19f, yScale2);
+}
+
+s32 func_800C0A04(void) {
+    s32 ret = func_800C07C4();
+    s32 var_v1;
+
+    if (ret != 0) {
+        D_801787AC = 255.0f;
+        D_801787B0 = 255.0f;
+        D_801787A4 += ret;
+
+        if (D_801787A4 >= 3) {
+            D_801787A4 = 0;
+            D_801787A4 = 0;
+        }
+
+        if (D_801787A4 < 0) {
+            D_801787A4 = 2;
+        }
+    }
+
+    if (Math_SmoothStepToF(&D_801787AC, D_801787B0, 0.5f, 100.0f, 1.0f) == 0.0f) {
+        if (D_801787B0 == 32.0f) {
+            D_801787B0 = 255.0f;
+        } else {
+            D_801787B0 = 32.0f;
+        }
+    }
+
+    return D_801787A4;
+}
+
+s32 func_800C0B0C(void) {
+    s32 i;
+    s32 colorGB[3];
+
+    RCP_SetupDL(&gMasterDisp, 0x4E);
+
+    for (i = 0; i < 3; i++) {
+        if (i == D_801787A4) {
+            colorGB[i] = D_801787AC;
+        } else {
+            colorGB[i] = 255;
+        }
+    }
+
+    gDPSetPrimColor(gMasterDisp++, 0, 0, 255, colorGB[0], colorGB[0], 255);
+    func_800BD05C(140.0f, 74.0f, 1.0f);
+    gDPSetPrimColor(gMasterDisp++, 0, 0, 255, colorGB[1], colorGB[1], 255);
+    func_800BD0B4(124.0f, 86.0f, 1.0f);
+    gDPSetPrimColor(gMasterDisp++, 0, 0, 255, colorGB[2], colorGB[2], 255);
+    func_800BD10C(144.0f, 154.0f, 1.0f);
+
+    return 0;
+}
+
+bool func_800C0C84(void) {
+    s32 i;
+
+    for (i = 0; i < 4; i++) {
+        if (gControllerPlugged[i] == 1) {
+            if (gControllerPress[i].button & A_BUTTON) {
```

Comment:
```suggestion
        if ((gControllerPlugged[i] == 1) && (gControllerPress[i].button & A_BUTTON)) {
```
these can probably be combined. Also that might want to be `== true`